File: core/src/main/java/vn/loitp/restapi/flickr/service/FlickrService.java
package vn.loitp.restapi.flickr.service;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;
import vn.loitp.restapi.flickr.model.photosetgetlist.WrapperPhotosetGetlist;
import vn.loitp.restapi.flickr.model.photosetgetphotos.WrapperPhotosetGetPhotos;

/**
 * @author loitp
 */

public interface FlickrService {
    @GET("rest/")
    Observable<WrapperPhotosetGetlist> photosetsGetList(@Query("method") String method,
                                                        @Query("api_key") String api_key,
                                                        @Query("user_id") String user_id,
                                                        @Query("page") int page,
                                                        @Query("per_page") int per_page,
                                                        @Query("primary_photo_extras") String primary_photo_extras,
                                                        @Query("format") String format,
                                                        @Query("nojsoncallback") int nojsoncallback);

    @GET("rest/")
    Observable<WrapperPhotosetGetPhotos> photosetsGetPhotos(@Query("method") String method,
                                                            @Query("api_key") String api_key,
                                                            @Query("photoset_id") String photoset_id,
                                                            @Query("user_id") String user_id,
                                                            @Query("extras") String extras,
                                                            @Query("per_page") int per_page,
                                                            @Query("page") int page,
                                                            @Query("format") String format,
                                                            @Query("nojsoncallback") int nojsoncallback);
}

File: core/src/main/java/com/loitpcore/restApi/flickr/service/FlickrService.kt
package com.loitpcore.restApi.flickr.service

import com.loitpcore.restApi.flickr.model.photoSetGetList.WrapperPhotosetGetlist
import com.loitpcore.restApi.flickr.model.photoSetGetPhotos.WrapperPhotosetGetPhotos
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by Loitp on 04,August,2022
 * Galaxy One company,
 * Vietnam
 * +840766040293
 * freuss47@gmail.com
 */
interface FlickrService {
    @GET("rest/")
    fun getListPhotoset(
        @Query("method") method: String?,
        @Query("api_key") apiKey: String?,
        @Query("user_id") userId: String?,
        @Query("page") page: Int,
        @Query("per_page") perPage: Int,
        @Query("primary_photo_extras") primaryPhotoExtras: String?,
        @Query("format") format: String?,
        @Query("nojsoncallback") noJsonCallback: Int
    ): Observable<WrapperPhotosetGetlist?>

    @GET("rest/")
    fun getPhotosetPhotos(
        @Query("method") method: String?,
        @Query("api_key") apiKey: String?,
        @Query("photoset_id") photosetId: String?,
        @Query("user_id") userId: String?,
        @Query("extras") extras: String?,
        @Query("per_page") perPage: Int,
        @Query("page") page: Int,
        @Query("format") format: String?,
        @Query("nojsoncallback") noJsonCallback: Int
    ): Observable<WrapperPhotosetGetPhotos?>
}
